designs/2024-ignore-file-directive/README.md

```diff
@@ -0,0 +1,121 @@
+
+This directive will work alongside the `.eslintignore` file in that it too will indicate which file be parsed or not. It extends the concept of ignoring source files by allowing files (specifically generated files) to exclude themselves.
+
+Lastly, library devlopers seldom actually mean "parse, but mute errors" when `/* eslint-disable */` is added at the top of files. While it the best option currently available, "completely skip this file" is actually meant.
+
+To summarize:
+
+1. Performance boost by not parsing unwanted files
+2. Don't crash over code the application developer has no/limited control over
+3. Does what leading `/* eslint-disable */` are meant to do
+
+## Detailed Design
+
+Please note that I am not familiar with the internals of ESLint and might make assumptions that are incorrect.
+
+For any _source file_ (the root of an AST), if it contains a leading `/* eslint-ignore-file */` directive, skip parsing it. More concretely, the following pseudocode algorithm explains how to determine whether a file should be ignored or processed:
+
+```
+for line of source file:
+  if line (is not zero or more whitespace) and (does not start with zero or more whitespace + "/"):
+    parse file
+  if line is "/* eslint-ignore-file */":
+    skip file parsing
+```
+
+A more sophisticated algorithm may be chosen at a later point in the lifecycle of this RFC.
+
+In practice, the imaginary
+
+```ts
+export declare function parseFile(filePath: string): AST;
+```
+
+needs to be replaced with:
+
+```ts
+export function parseFile(filePath: string): AST | null {
+  const fileContent = streamFileContent(filePath);
+
+  if (shouldIgnore(fileContent)) return null;
+  return parseFileImpl(fileContent.collectToString());
+}
+
+declare function parseFileImpl(fileContent: string): AST;
+```
+
+How the rest of the system deals with `null` results from parsing a source file is not in the scope of this RFC.
+
+If an `/* eslint-ignore-file */` directive is encountered while parsing a file, a non-fatal error shall be raised, and the directive shall have no further effect.
+
+## Documentation
+
+TBD. The primary audience is a library author. The point **Drawbacks > Bardwards Compatibility** shall be highlighted.
+
+## Drawbacks
+
+1. Backwards Compatibility
+
+Library authors will need to prefix their generated code with both `/* eslint-ignore-file */` and `/* eslint-disable */` directives in order to not break existing projects that have not upgraded their ESLint core to a version that supports the `/* eslint-ignore-file */` directive. Otherwise, updating libraries without updating ESLint accordingly will lead to possible linging errors on generated source files that were previously muted.
+
+2. Primitive Parser
```

Comment:
Why would we parse at all? Why not just use a regular expression to see if the text exists in the source?